src/item.cpp

```diff
@@ -2252,6 +2256,10 @@ std::string item::tname( unsigned int quantity, bool with_prefix ) const
     if (has_flag("FIT")) {
         ret << _(" (fits)");
     }
+    
+    if( has_flag( "FILTHY" ) && g->u.has_trait( "SQUEAMISH" ) ) {
+        ret << _(" (filthy)" );
+    }
```

Comment:
As a final step I'd wrap this into a function and reuse it three times. Something like:
```c++
bool item::is_disgusted_by( const player &p ) const {
    return has_flag( "FILTHY" ) && p.has_trait( "SQUEAMISH" );
}
```